### ISSUE TITLE
login() API provides no way for caller to investigate error code/msg

### ISSUE THREAD

**Issue by pmorch** [mrproper/proxmox-ve-api-perl#3]
The canonical way to login is to:

```
$host->login() or die ('Couldnt log in to proxmox host');
```

But what exactly failed? The caller has no idea. `login()` should return an error that could be investigated or provide an error() API call for that purpose so proper diagnosis is possible without having to revert to 

```
vi lib/Net/Proxmox/VE/Access.pm
```

and printf...

If login() changed and returned false on OK, but returned the `HTTP::Response / $reponse` on error, one could:

```
if (my $error = $host->login()) {
    die sprintf "Couldn't login. HTTP code: %d status line: %s",
        $error->code(),
        $error->status_line
}
```

(In our case,

```
printf("Code: %d\nStatus Line: %s\n",
    $response->code,
    $response->status_line
);
```

printed:

```
Code: 500
Status Line: 500 Can't connect to proxmox01:8006 (certificate verify failed)
```

but that is a separate issue - more on that later)

**Comment by pmorch** [comment]
(And the "separate issue" is addressed in merge request #4 by [this commit](https://github.com/pmorch/proxmox-ve-api-perl/commit/8e86ee49ee4a5bd59b90804f405f961809cf580f))

**Comment by mrproper** [comment]
Is there still anything left on this?
